mk/config.mk

```diff
@@ -35,11 +35,15 @@ WARNS ?= 3
 # Define NOWERROR=1 so that warnings are not treated as errors
 # NOWERROR=1
 
-# Define DEBUG=1 to compile with -g option
+# Define DEBUG=1 to compile without optimization (forces -O0)
 # DEBUG=1
 
-# If 1, debug mode of the tee firmware (CPU restart, Core Status)
-CFG_TEE_CORE_DEBUG ?= 0
+# If y, enable debug mode of the tee firmware (CPU restart, Core Status
+# verbose, panic & assert verbose). When disable, NDEBUG directive is defined.
+CFG_TEE_CORE_DEBUG ?= n
+ifneq ($(CFG_TEE_CORE_DEBUG),y)
+NDEBUG ?= 1
```

Comment:
There is currently no easy way to set `NDEBUG` when building opee core.
Hence i think it is nice to rely on `CFG_TEE_CORE_DEBUG.`

maybe we could discard setting `NDEBUG` from optee config and change source files using `NDEBUG` to also rely on `CFG_TEE_CORE_DEBUG:`
```
lib/libutils/isoc/include/assert.h
-#ifdef NDEBUG
+#if defined(NDEBUG) || defined(CFG_TEE_CORE_DEBUG)
 #define assert(expr)	do { } while (0)
 #else
 #define assert(expr) \
```